src/system/SystemMutex.h

```diff
@@ -78,8 +84,8 @@ class DLL_EXPORT Mutex
 #if (configSUPPORT_STATIC_ALLOCATION == 1)
     StaticSemaphore_t mFreeRTOSSemaphoreObj;
 #endif // (configSUPPORT_STATIC_ALLOCATION == 1)
-    volatile SemaphoreHandle_t mFreeRTOSSemaphore;
-    volatile int mInitialized;
+    volatile SemaphoreHandle_t mFreeRTOSSemaphore = nullptr;
```

Comment:
Why not only ever use static semaphores on FreeRTOS?

Also, what about Zephyr?

Reply:
if using StaticSemaphore_t, in esp32, I am seeing invalid cast from type 'StaticSemaphore_t' {aka 'xSTATIC_QUEUE'} to type 'QueueHandle_t' {aka 'void*'}
     xSemaphoreGive(this->mFreeRTOSSemaphore)，
in espe32, i see typedefQueueHandle_tSemaphoreHandle_t

Reply:
Why not try to resolve it rather than having two implementations for the same RTOS. I don't recall offhand, but I think static semaphore type is always supported. @dhrishi for ESP feedback.

Overall, instead of many #ifdefs in core code, why isn't this SystemMutex abstraction just implemented per platform as opposed to once non-portably withany ifdefs?

Reply:
Fixing semaphores should probably be a separate PR

Reply:
Create https://github.com/project-chip/connectedhomeip/issues/7415 for issue track